tests/gpts/gitlab/suites-config/gitlab_advanced_jenkins.ts

```diff
@@ -0,0 +1,290 @@
+import { beforeAll, expect, it, describe } from "@jest/globals";
+import { DeveloperHubClient } from "../../../../src/apis/backstage/developer-hub";
+import { TaskIdReponse } from "../../../../src/apis/backstage/types";
+import { GitLabProvider } from "../../../../src/apis/git-providers/gitlab";
+import { Kubernetes } from "../../../../src/apis/kubernetes/kube";
+import { generateRandomChars } from "../../../../src/utils/generator";
+import { checkComponentSyncedInArgoAndRouteIsWorking, checkEnvVariablesGitLab, cleanAfterTestGitLab, createTaskCreatorOptionsGitlab, getDeveloperHubClient, getGitLabProvider, getJenkinsCI, getRHTAPRootNamespace} from "../../../../src/utils/test.utils";
+import { JenkinsCI } from "../../../../src/apis/ci/jenkins";
+import { Utils } from "../../../../src/apis/git-providers/utils";
+
+/**
+ * 1. Components get created in Red Hat Developer Hub
+ * 2. Check that components gets created successfully in Red Hat Developer Hub
+ * 3. Check if Red Hat Developer Hub created GitLab repositories with Jenkinsfiles
+ * 4. Commit Jenkins agent settings and enable ACS
+ * 5. Creates job in Jenkins
+ * 6. Trigger Jenkins Job and wait for finish 
+ * 7. Perform an commit in GitLab
+ * 8. Trigger Jenkins Job and wait for finish
+ * 9. Check if the application is deployed in development namespace and pod is synched
+ * 
+ * @param softwareTemplateName The name of the software template.
+ */
+export const gitLabJenkinsAdvancedTests = (softwareTemplateName: string, stringOnRoute: string) => {
+    describe(`Red Hat Trusted Application Pipeline ${softwareTemplateName} GPT tests GitLab provider with public/private image registry with promotion pipeline`, () => {
+
+        let backstageClient: DeveloperHubClient;
+        let developerHubTask: TaskIdReponse;
+        let gitLabProvider: GitLabProvider;
+        let kubeClient: Kubernetes;
+        let jenkinsClient: JenkinsCI;
+
+        let gitlabRepositoryID: number;
+        let gitlabGitOpsRepositoryID: number;
+        let gitopsPromotionMergeRequestNumber: number;
+        let RHTAPRootNamespace: string;
+
+        const componentRootNamespace = process.env.APPLICATION_ROOT_NAMESPACE || 'rhtap-app';
+        const developmentEnvironmentName = 'development';
+        const stagingEnvironmentName = 'stage';
+        const productionEnvironmentName = 'prod';
+        const developmentNamespace = `${componentRootNamespace}-${developmentEnvironmentName}`;
+        const stageNamespace = `${componentRootNamespace}-${stagingEnvironmentName}`;
+        const prodNamespace = `${componentRootNamespace}-${productionEnvironmentName}`;
+
+
+        const gitLabOrganization = process.env.GITLAB_ORGANIZATION || '';
+        const repositoryName = `${generateRandomChars(9)}-${softwareTemplateName}`;
+
+        const quayImageName = "rhtap-qe";
+        const quayImageOrg = process.env.QUAY_IMAGE_ORG || '';
```

Comment:
@jkopriva here also can we keep these image variable names general, not specific to quay?
```suggestion
        const imageName = "rhtap-qe";
        const imageOrg = process.env.QUAY_IMAGE_ORG || '';
```

Reply:
fixed, thanks!. resolving.